### DISCUSSION TITLE
Phase field in axisymmetric coordinates

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#24073] [Q&A General] [answered]
Hello everyone,

Based on the way that phase field kernels in the cahn hilliard split form are developed, can I use them in axisymmetric coordinates or custom kernels must be developed like the navier stokes equation in rz coordinates?

Thanks,
Ali

**Comment by jiangwen84** [comment, marked as answer]
If you are using split forms (into two second-order equation), kernels will be the same. If you are not, then you need a special kernel for the 4th- order term.

**Reply by alimostafavi24** [reply]
I am using the split form and only deal with 2nd order derivative.

But are you 100% sure about this? This is the definition of the 1st and 2nd order derivatives of a scalar variable and this completely different with xy coordinates.
![Screenshot from 2023-04-14 14-38-42](https://user-images.githubusercontent.com/119982603/232163433-da8c52e3-ef0f-4f16-a5d7-f1edc6a0aae7.png)

**Comment by alimostafavi24** [comment]
I think that phase field cannot be used in axisymmetric coordinates. Because the 2nd order derivative of c is still present in the weak form residual and the definition of which is very different in XY and RZ coordinates. Can someone please confirm that I am correct or not?


![Screenshot from 2023-04-15 12-23-48](https://user-images.githubusercontent.com/119982603/232249828-acd7805e-1413-46df-8576-d2e4954816a6.png)




Thanks,
Ali

**Reply by hugary1995** [reply]
In RZ we assume the derivatives in theta to be zero. With some work you can prove that the divergence theorem holds in cylindrical coordinates, using which you can arrive at the same weak form. Transforming the integration from cylindrical coordinates to Cartesian requires simply replacing `r` with `x`, and an additional coordinate transformation factor `r` which is accounted for for you in `_coord[_qp]`.

**Reply by jiangwen84** [reply]
I think now I understand your concern. In the plot you show, the residual term has \Lap c in the ```SplitCHParsed``` kernel. The description here is a bit confusing. In the actual code,  the \Lap c will be transformed into the \nabla c by taking the integral by part, so only gradient term will be remained. See the code below
```
Real
SplitCHCRes::computeQpResidual()
{
  Real residual =
      SplitCHBase::computeQpResidual(); //(f_prime_zero+e_prime)*_test[_i][_qp] from SplitCHBase

  residual += -_w[_qp] * _test[_i][_qp];
  residual += _kappa[_qp] * _grad_u[_qp] * _grad_test[_i][_qp];

  return residual;
}
```

After you understand this, you can follow @hugary1995 's comment to prove the split form will work for RZ coordinate system.

**Reply by alimostafavi24** [reply]
Yes, exactly! Documentation on the split phase field equations is a bit misleading here! Thank you very much for the help (@jiangwen84 @hugary1995 ).
